### PR TITLE
JIT: remove size contribution to perf score

### LINKED ISSUE #96878: [JIT] PerfScore includes code size impact?
https://github.com/dotnet/runtime/blob/64fe78edb3fe1f02c5c3998e39491a9e100c139f/src/coreclr/jit/codegencommon.cpp#L2005-L2011

This was part of the original design https://github.com/dotnet/coreclr/pull/26456 but I don't think it makes sense, either from a "units" standpoint (speed is not the same as size, and existence of a widely valid conversion ratio is not obvious) or generally.

And I think it may also be complicating attempts to leverage perf score for learning ,eg https://github.com/dotnet/runtime/issues/92915#issuecomment-1744042491

We have a separate size metric, so in cases where we care about both, we can just look at that as well.

### PR DESCRIPTION
**Pull request by AndyAyersMS**
Keep the perf score a pure performance metric. Size is emitted as a separate metric, so clients interested in a metric that is some combination of size and speed can do their own computations based on these two values.

Resolves #96878.

### CODE REVIEW

**Comment by AndyAyersMS** [comment]
@dotnet/jit-contrib FYI
@jakobbotsch PTAL

**Review by jakobbotsch — APPROVED** [review]

File: src/coreclr/jit/codegencommon.cpp [hunk 1]
```diff
@@ -2002,14 +2002,6 @@ void CodeGen::genEmitMachineCode()
     compiler->compCodeGenDone = true;
 #endif
 
-#if defined(DEBUG) || defined(LATE_DISASM)
-    // Add code size information into the Perf Score
-    // All compPerfScore calculations must be performed using doubles
-    compiler->info.compPerfScore += ((double)compiler->info.compTotalHotCodeSize * (double)PERFSCORE_CODESIZE_COST_HOT);
-    compiler->info.compPerfScore +=
-        ((double)compiler->info.compTotalColdCodeSize * (double)PERFSCORE_CODESIZE_COST_COLD);
```
Comment (BruceForstall) [inline src/coreclr/jit/codegencommon.cpp:2010]: You should remove the definitions of `PERFSCORE_CODESIZE_COST_HOT` / `PERFSCORE_CODESIZE_COST_COLD`, which will be no longer used.

**Comment by BruceForstall** [comment]
Odd the CI never ran anything...

File: src/coreclr/jit/codegencommon.cpp (hunk 1 above)
Comment (AndyAyersMS) [reply src/coreclr/jit/codegencommon.cpp:2010]: thanks, done.

**Review by BruceForstall — APPROVED** [review]

**Comment by AndyAyersMS** [comment]
NAOT failure was #94728.

**Pull request merged**